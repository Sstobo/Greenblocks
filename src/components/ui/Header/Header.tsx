import React from 'react';
import {useState} from 'react';

// Help find tailwind classnames as we go
const Header = (props) => {
    const [count, setCount] = useState(0);

    const heading: string = props?.heading || '';
    return (
        <div className="container mx-auto p-8">
            <h6 data-testid="subheader" className="text-2xl font-bold text-center text-gray-800">Hello World</h6>
            <h1 className="text-4xl font-bold text-center text-gray-800">{heading}</h1>
            <p>{count}</p>

            <button onClick={() => setCount(count + 1)}>Click me</button>
        </div>
    )
}

export default Header