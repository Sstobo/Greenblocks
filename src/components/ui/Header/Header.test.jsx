import React from 'react';
import renderer from 'react-test-renderer';
import Header from './Header';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';



describe('Header', () => {
// use renderer for snapshots
  it('should render correctly into the dom', () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  it("subheader should say Hello World", () => {
    const { getByTestId } = render(<Header />);
    const subheader = getByTestId('subheader');
    expect(subheader).toHaveTextContent('Hello World');
  });

  it('should show proper initial count', () => {
    const { getByText } = render(<Header />);
    expect(getByText('0')).toBeInTheDocument();
  });

  it('should increment count by 1 when button is clicked', () => {
    const { getByText } = render(<Header />);
    const button = getByText('Click me');
    fireEvent.click(button);
    expect(getByText('1')).toBeInTheDocument();
  });

  it('should increment count by 3 when button is clicked 3 times', () => {
    const { getByText } = render(<Header />);
    const button = getByText('Click me');
    for (let i = 0; i < 3; i++) {
      fireEvent.click(button);
    }
    expect(getByText('3')).toBeInTheDocument();
  });
});