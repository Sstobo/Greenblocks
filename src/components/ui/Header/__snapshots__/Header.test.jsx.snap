// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly into the dom 1`] = `
<div
  className="container mx-auto p-8"
>
  <h6
    className="text-2xl font-bold text-center text-gray-800"
    data-testid="subheader"
  >
    Hello World
  </h6>
  <h1
    className="text-4xl font-bold text-center text-gray-800"
  />
  <p>
    0
  </p>
  <button
    onClick={[Function]}
  >
    Click me
  </button>
</div>
`;
